orcoursetrion/lib/github.py

```diff
@@ -390,3 +393,59 @@ def delete_web_hooks(self, org, repo):
                 raise GitHubUnknownError(response.text)
             num_hooks_removed += 1
         return num_hooks_removed
+
+    def shallow_copy_repo(self, src_repo, dst_repo, committer, branch=None):
+        """Copies one branch repo's contents to a new repo in the same
+        organization without history.
+
+        .. DANGER::
+           This will overwrite the destination repo's default branch and
+           rewrite its history.
+
+        The basic workflow is:
+
+        - Clone source repo
+        - Remove source repo ``.git`` folder
+        - Initialize as new git repo
+        - Set identity
+        - Add everything and commit
+        - Force push to destination repo
+
+        Args:
+            src_repo (str): Full git url to source repo.
+            dst_repo (str): Full git url to destination repo.
+            committer (dict): {'name': ..., 'email': ...} for the name
+                and e-mail to use in the initial commit of the
+                destination repo.
+            branch (str): Option branch, if not specified default is used.
+        Raises:
+            sh.ErrorReturnCode
+        Returns:
+            None
+
+        """
+        # Grab current working directoy so we return after we are done
+        cwd = unicode(sh.pwd().rstrip('\n'))
+        tmp_dir = tempfile.mkdtemp(prefix='orc_git')
+        try:
+            sh.cd(tmp_dir)
+            if branch is None:
+                sh.git.clone(src_repo, depth=1)
+            else:
+                sh.git.clone(src_repo, depth=1, branch=branch)
+            sh.cd(sh.glob('*')[0])
```

Comment:
I'll add a comment to make it clear since it is a bit magic here, but there isn't a race condition here because this is happening inside the `tempfile.mkdtemp` which guarantees a unique folder to work on.  This really just saves doing a string parse on the git url that is prone to mistakes and I have a fairly strong guarantee that the git repo is the only folder in the newly minted and soon to be deleted temp directory.